implementations/rust/ockam/ockam_transport_tcp/src/transport/common.rs

```diff
@@ -154,6 +165,116 @@ pub(super) fn parse_socket_addr(s: &str) -> Result<SocketAddr> {
     Ok(s.parse().map_err(|_| TransportError::InvalidAddress)?)
 }
 
+/// Connect to a socket address via a regular TcpStream
+#[instrument(skip_all)]
+pub(crate) async fn connect(socket_address: SocketAddr) -> Result<(OwnedReadHalf, OwnedWriteHalf)> {
+    Ok(create_tcp_stream(socket_address).await?.into_split())
+}
+
+/// Create a TCP stream to a given socket address
+pub(crate) async fn create_tcp_stream(socket_address: SocketAddr) -> Result<TcpStream> {
+    debug!(addr = %socket_address, "Connecting");
+    let connection = match TcpStream::connect(socket_address).await {
+        Ok(c) => {
+            debug!(addr = %socket_address, "Connected");
+            c
+        }
+        Err(e) => {
+            debug!(addr = %socket_address, err = %e, "Failed to connect");
+            return Err(TransportError::from(e))?;
+        }
+    };
+
+    let mut keepalive = TcpKeepalive::new()
+        .with_time(Duration::from_secs(300))
+        .with_interval(Duration::from_secs(75));
+
+    cfg_if! {
+        if #[cfg(unix)] {
+           keepalive = keepalive.with_retries(2);
+        }
+    }
+
+    let socket = SockRef::from(&connection);
+    socket
+        .set_tcp_keepalive(&keepalive)
+        .map_err(TransportError::from)?;
+
+    Ok(connection)
+}
+
+/// Connect to a socket address via a TlsStream
+#[allow(clippy::type_complexity)]
+#[instrument(skip_all)]
+pub(crate) async fn connect_tls(
+    hostname_port: &HostnamePort,
+) -> Result<(
+    ReadHalf<TlsStream<TcpStream>>,
+    WriteHalf<TlsStream<TcpStream>>,
+)> {
+    // Don't try to establish a TLS connection locally
+    if ["127.0.0.1", "localhost"].contains(&hostname_port.hostname().as_str()) {
```

Comment:
what about IPv6?

Reply:
Funny that you see this, I have removed those lines since.